platform/admin/core/admin-core-logviewer/src/main/webapp/js/reducers/index.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+
+var m = require('merge')
+
+// total number of log entries to keep in memory that a
+// user can filter through
+var MAX_FILTER_ENTRIES = 5000
+
+module.exports = function (state, action) {
+  if (state === undefined) {
+    return {
+      logs: [],
+      filter: { level: 'ALL' },
+      open: true,
+      displaySize: 25 // number of rows to display
+    }
+  }
+
+  switch (action.type) {
+    case 'FILTER_LOGS':
+      return m(true, state, {
+        displaySize: 25,
+        filter: m(true, state.filter, action.filter)
```

Comment:
These reducers would be easier to read/maintain using the ES6 spread operator which we could get if we use Babel.js.

http://redux.js.org/docs/recipes/UsingObjectSpreadOperator.html
https://babeljs.io/